libraries/chain/controller.cpp

```diff
@@ -1301,6 +1303,9 @@ struct controller_impl {
          if( shutdown ) shutdown();
       } );
 
+      sync_call_wasm_alloc.resize(16); // Will change to use max_sync_call_depth of global property object in future version
```

Comment:
Seems like this would need to be resized on each read only thread too?

Reply:
It might make more sense to do more of that work on #1257  -- it would have to resize upon the parameters changed etc

Reply:
Yes, on #1257, I will do the resize changes using the new parameters.

Reply:
I added the initialization of sync_call_wasm_alloc on read threads at start up https://github.com/AntelopeIO/spring/pull/1251/commits/06bdd12b595735e7a885e3422925037f5d58eaa2. Will update it in #1257 with actual sync_call_depth.

Reply:
I created a task for resizing the vector as it requires some experiment. https://github.com/AntelopeIO/spring/issues/1269

Reply:
Probably also dependent on if we allow privileged sync calls, since a privileged sync call might, for example, reduce the sync call depth beyond the current depth